training/CHAPEL_TUTORIAL.md

```diff
@@ -993,358 +1073,635 @@ Expected Output:
+    writeln("src: ", src);
+    writeln("dst: ", dst);
+}
+else {
+    // when the source is remote we use a srcAggregator
+    forall (d, i) in zip(dst, 0..) with (var agg = new SrcAggregator(int)) {
+      // locNum is which locale this loop iteration is executing on
+      var locNum = here.id;
+      // localSubDom is the chunk of the distributed arrays that live on this locale
+      var localSubDom = dom.localSubdomain();
+      
+      // we use a single writeln to avoid interleaving output from another locale 
+      writeln("\niteration num: ", i, "\n  on Locale: ", locNum,
+              "\n  on localSubDom: ", localSubDom, "\n  src[", (i + 3) % 6, "] is remote",
+              "\n  dst[", i, "] is local");
+    
+      // since src is remote, we use a src aggregator
+      // assignment without aggregation would look like:
+      // d = src[ (i + 3) % 6 ]
+      agg.copy(d, src[ (i + 3) % 6 ]);
     }
+    writeln();
+    writeln("src: ", src);
+    writeln("dst: ", dst);
 }
+```
 
-config param SIZE = 1000000;
-const distDom = Block.createDomain({0..#SIZE});
+using a `DstAggregator`:
+```console
+$ chpl tutorial.chpl --no-cache-remote
+$ ./tutorial -nl 2
 
-// create a array containing tuples of uints
-var hashes: [distDom] (uint, uint) = (1, 1):(uint, uint);
+src: 0 1 2 3 4 5
+dst: 0 0 0 0 0 0
+
+iteration num: 0
+  on Locale: 0
+  on localSubDom: {0..2}
+  src[0] is local
+  dst[3] is remote
+
+iteration num: 2
+  on Locale: 0
+  on localSubDom: {0..2}
+  src[2] is local
+  dst[5] is remote
+
+iteration num: 1
+  on Locale: 0
+  on localSubDom: {0..2}
+  src[1] is local
+  dst[4] is remote
+
+iteration num: 3
+  on Locale: 1
+  on localSubDom: {3..5}
+  src[3] is local
+  dst[0] is remote
+
+iteration num: 5
+  on Locale: 1
+  on localSubDom: {3..5}
+  src[5] is local
+  dst[2] is remote
+
+iteration num: 4
+  on Locale: 1
+  on localSubDom: {3..5}
+  src[4] is local
+  dst[1] is remote
+
+src: 0 1 2 3 4 5
+dst: 3 4 5 0 1 2
+```
+
+using a `SrcAggregator`:
+```console
+$ ./tutorial -nl 2 --UseDstAgg=false
+src: 0 1 2 3 4 5
+dst: 0 0 0 0 0 0
+
+iteration num: 0
+  on Locale: 0
+  on localSubDom: {0..2}
+  src[3] is remote
+  dst[0] is local
+
+iteration num: 1
+  on Locale: 0
+  on localSubDom: {0..2}
+  src[4] is remote
+  dst[1] is local
+
+iteration num: 2
+  on Locale: 0
+  on localSubDom: {0..2}
+  src[5] is remote
+  dst[2] is local
+
+iteration num: 3
+  on Locale: 1
+  on localSubDom: {3..5}
+  src[0] is remote
+  dst[3] is local
+
+iteration num: 5
+  on Locale: 1
+  on localSubDom: {3..5}
+  src[2] is remote
+  dst[5] is local
+
+iteration num: 4
+  on Locale: 1
+  on localSubDom: {3..5}
+  src[1] is remote
+  dst[4] is local
+
+src: 0 1 2 3 4 5
+dst: 3 4 5 0 1 2
+```
 
-var upperEntry = new SymEntry(SIZE, uint);
-var lowerEntry = new SymEntry(SIZE, uint);
+<a id="perf"></a>
+## Performance and Diagnostics
+Now that you're a pro at writing chapel code, let's talk about how to make the code you write more efficient!
```

Comment:
This makes me think:  What do you think about writing your aggregator examples without the aggregators first, to show the simpler version of the computation you're expressing followed by the more complex (but efficient) form?  (in which case, maybe this mention of efficiency should come just before showing the aggregators).